### PR TITLE
[QUAD-196]: Added the ability to view all the public networks in quadriga.

### PR DESCRIPTION
**Pull request by suprajkolluri**
[QUAD-196]: Added the ability to view all the public networks in quadriga in a single page in quadriga/sites section.

### CODE REVIEW

**Comment by suprajkolluri** [comment]
@jdamerow Please review. Also I have squashed my commits to make the commit history look clean :)

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkManager.java [hunk 1]
```diff
@@ -349,6 +349,25 @@ public INetwork getNetwork(String networkId) throws QuadrigaStorageException {
...
+    */
+    @Override
+    @Transactional
+    public List<INetwork> getApprovedNetworkList() throws QuadrigaStorageException {
+        List<INetwork> networkList = new ArrayList<INetwork>();
+
+        try {
+            networkList = networkmapper.getListOfApprovedNetworks();
+        } catch (QuadrigaStorageException e) {
+            logger.error("Something went wrong in DB", e);
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkManager.java:368]: the method signature says it throws an exception but here you just log it and don't throw anything.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/open/NodeInfoController.java [hunk 2]
```diff
@@ -26,26 +26,27 @@
...
+            throws Exception {
         Set<TextOccurance> occurances = networkManager.getTextsForConceptId(conceptId, ETextAccessibility.PUBLIC);
         JSONArray projectTexts = new JSONArray();
         JSONArray otherProjectsTexts = new JSONArray();
-        
+
         for (TextOccurance occur : occurances) {
-            if (occur.getProject().getUnixName().equals(projectUnix.trim())) {
+            if (occur.getProject().getUnixName().equals(projectUnix.trim())
+                    || occur.getProject().getUnixName().equals("unkown")) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/open/NodeInfoController.java]: where is this coming from? why is this checking for a project with unix name "unkown" (which is misspelled btw).

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java [hunk 3]
```diff
@@ -69,6 +69,27 @@ public ITransformedNetwork getTransformedNetwork(String networkId) throws Quadri
 
         return networkJSon;
     }
+    
+    @Override
+    public ITransformedNetwork getTransformedApprovedNetworks(List<INetwork> networkList) throws QuadrigaStorageException {
+
+        ITransformedNetwork networkJSon = null;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java]: declare where it's actually needed

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java [hunk 4]
```diff
@@ -69,6 +69,27 @@ public ITransformedNetwork getTransformedNetwork(String networkId) throws Quadri
...
+
+        List<INetworkNodeInfo> networkTopNodesList = new ArrayList<INetworkNodeInfo>();
+        for(INetwork curnetwork: networkList){
+            try {
+                List<INetworkNodeInfo> curTopNodesList = networkManager.getNetworkTopNodes(curnetwork.getNetworkId());
+                networkTopNodesList.addAll(curTopNodesList) ;
+            } catch (QuadrigaStorageException e) {
+                logger.error("DB Error while getting network top nodes", e);
+                return null;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java]: same thing here: method declares an exception but non is thrown.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java [hunk 5]
```diff
@@ -69,6 +69,27 @@ public ITransformedNetwork getTransformedNetwork(String networkId) throws Quadri
...
+            }
+        }
+        
+        networkJSon = transformer.transformNetwork(networkTopNodesList);
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java]: just return directly in this line

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/mapper/impl/NetworkMapper.java [hunk 6]
```diff
@@ -130,6 +130,16 @@ public INetwork getNetworkShallowDetails(NetworksDTO networksDTO) throws Quadrig
     }
     
     @Override
+    public List<INetwork> getListOfApprovedNetworks() throws QuadrigaStorageException {
+        List<INetwork> networkList = null;
+        List<NetworksDTO> networksDTO = dbconnect.getApprovedNetworkList();
+        if(networksDTO!=null){
+            networkList = getNetworkListFromDTOList(networksDTO);
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/mapper/impl/NetworkMapper.java]: return here

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/open/NodeInfoController.java (hunk 2 above)
Comment (suprajkolluri) [reply Quadriga/src/main/java/edu/asu/spring/quadriga/rest/open/NodeInfoController.java]: This is coming from the UI, views/public/sites/publicnetworks.jsp line 147.
Corrected the spelling in the jsp also

**Pull request closed without merging**